apis/v1/shared_types.go

```diff
@@ -743,3 +743,115 @@ const (
 	// Support: Implementation-specific
 	NamedAddressType AddressType = "NamedAddress"
 )
+
+// SessionPersistence defines the desired state of
+// SessionPersistence.
+type SessionPersistence struct {
+	// SessionName defines the name of the persistent session token
+	// which may be reflected in the cookie or the header. Users
+	// should avoid reusing session names to prevent unintended
+	// consequences, such as rejection or unpredictable behavior.
+	//
+	// Support: Implementation-specific
+	//
+	// +optional
+	// +kubebuilder:validation:MaxLength=4096
```

Comment:
This seems like an awfully long length, what kind of session name would be this long?

Reply:
Good question...

I believe I got the max length from https://www.rfc-editor.org/rfc/rfc6265:
>   Practical user agent implementations have limits on the number and
   size of cookies that they can store.  General-use user agents SHOULD
   provide each of the following minimum capabilities:
>
>    o  At least 4096 bytes per cookie (as measured by the sum of the
      length of the cookie's name, value, and attributes).

But this is the _minimum capabilities_, so I suppose that's the worst case (smallest supported cookie size)...of the worst case (the entire cookie size is the name...).

So yea 4096 is a somewhat arbitrary. Is it best practice to limit string lengths? Or is it okay to leave unspecified?

Reply:
>Is it best practice to limit string lengths?

Yep, it's been a strong requirement to avoid any kind of unbounded values in Gateway API.

> as measured by the sum of the length of the cookie's name, value, and attributes

In the case of cookie-based persistence this would just be the cookie name right? (Not the value or attributes)

Reply:
> In the case of cookie-based persistence this would just be the cookie name right? (Not the value or attributes)

Correct. `SessionName` is just the cookie name. I was just trying to find established limits, and the only one I came across was this 4096 bytes value for the *whole* cookie.

Since I can't find any enforced value, should we start with something reasonable? Maybe 128 or 256?

Reply:
Let's start with 128 and increase if/when necessary. It's hard for me to think of a use case that would require a longer session name, but I could be missing something.